includes/payment-methods/class-upe-payment-gateway.php

```diff
@@ -551,9 +569,16 @@ public function process_payment( $order_id ) {
 						$request->set_payment_method_options( $payment_method_options );
 					}
 					$updated_payment_intent = $request->send( 'wcpay_update_intention_request', $order, $payment_intent_id );
-				} catch ( Amount_Too_Small_Exception $e ) {
-					// This code would only be reached if the cache has already expired.
-					throw new Exception( WC_Payments_Utils::get_filtered_error_message( $e ) );
+				} catch ( Exception $e ) {
+					try {
+						$intent = Get_Intention::create( $payment_intent_id )->send( 'wcpay_get_intention_request' );
+						$charge = $intent->get_charge();
+						$this->order_service->mark_payment_failed( $order, $payment_intent_id, $intent->get_status(), $charge ? $charge->get_id() : '', $e->getMessage() );
```

Comment:
Note: that adds extra order notes, exposing the original issue and cases when APIs are down.

Reply:
From support PoV I believe it's a good thing to have, but I would love to hear more thoughts on this change.